public/js/actions/subscriptions.js

```diff
@@ -98,11 +100,20 @@ export function _createSubscription({dispatch, productId,
     console.log('Successfully subscribed + completed payment');
     dispatch(transactionActions.complete({userEmail: email}));
   }).fail($xhr => {
-    if (data.pay_method_nonce) {
-      dispatch({
-        type: actionTypes.CREDIT_CARD_SUBMISSION_ERRORS,
-        apiErrorResult: $xhr.responseJSON,
-      });
+    console.log($xhr);
+    if ($xhr.status === 400 && $xhr.responseJSON &&
+        $xhr.responseJSON.error_response) {
+      var allErrors = $xhr.responseJSON.error_response.__all__;
+      if (allErrors && hasErrorSubString(allErrors, 'already subscribed')) {
+        dispatch(appActions.error('Subscription creation failed', {
+          userMessage: gettext('User is already subscribed to this product'),
+        }));
+      } else if (data.pay_method_nonce) {
+        dispatch({
+          type: actionTypes.CREDIT_CARD_SUBMISSION_ERRORS,
+          apiErrorResult: $xhr.responseJSON,
```

Comment:
The difference is that card submission errors are much lower level. Whilst it's the same data it doesn't result in a hard failure. It's things like field errors.